producers/smd_producer.py

```diff
@@ -326,11 +393,21 @@ def get_sd_file(write_dir, exp, hutch):
 	logger.debug('Could not find {0} in list of available hutches'.format(hutch))
 	sys.exit()	
 
+# Figure out where we are and where to look for data
 xtc_files = []
-# If experiment matches, check for files in ffb
-useFFB=False
-#with the new FFB, no need to check both on & offline as systems are independant.
-if hostname.find('drp')>=0:
+useFFB = False
+onS3DF = False
+onPSANA = False
+
+if hostname.find('sdf')>=0:
+    logger.debug('On S3DF')
+    onS3DF = True
+    if 'ffb' in PSDM_BASE.as_posix():
+        useFFB = True
+        # do we need to do smth to wait for
+        # files?
```

Comment:
We are waiting below, I'm not sure what is meant with the comment?

Reply:
Only in the case where we run from the drp hosts. The question here is exactly that: do we want to have something similar? In practice I don't remember seeing that we have to wait for files. In fact if we do, it will likely mess with the elog API call to get the right file location when running on S3DF (S3DF vs the FFB mount)

Reply:
Yeah, sounds about right.